### PR TITLE
Export API

### PR DESCRIPTION
**Pull request by ivkean**
#### Status

READY

#### What's this PR do?

- [x] Partners
- [x] Vouchers
- [x] Orders
- [x] Deals

#### Where should the reviewer start?
- config/api/api.go
- helper/middlewares.go
- main.go

#### What are the relevant Git/YouTrack tickets?

// Put in link to Git Issue or YouTrack ticket

#### Questions:

- Is there a blog post?
- Does the knowledge base need an update?
- Does this add new dependencies which need to be added to?

### CODE REVIEW

File: helper/middlewares.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+func AuthRequired() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		header := c.Request.Header.Get("Authorization")
+		if header == "" || !strings.Contains(header, "Basic") {
+			c.AbortWithStatus(403)
+		}
+
+		value := strings.Split(header, " ")
+		token := value[1]
+		sEnc := b64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", os.Getenv("ADMIN_USERNAME"), os.Getenv("ADMIN_PASSWORD"))))
```
Comment (hieuphq) [inline helper/middlewares.go]: calm down guy! Are you serious?

**Comment by ivkean** [comment]
@hieuphq what's wrong ? Please add more details so that everyone can get what you mean

File: helper/middlewares.go (hunk 1 above)
Comment (hieuphq) [inline helper/middlewares.go]: @ivkean how to know your key? Front-end must get your admin's user name and password, parse to a string(base64). Is this safe when you do that?

**Comment by hieuphq** [comment]
👍

**Pull request merged**